### PR TITLE
[backport][release/v1.23] connection handler: correct the handling of listener update 

### PR DESCRIPTION
**Pull request by soulxu**
Signed-off-by: He Jie Xu <hejie.xu@intel.com>

Commit Message: connection handler: correct the handling of listener update
Additional Description:
This is cherry-pick from https://github.com/envoyproxy/envoy/pull/22622

Risk Level: high
Testing: unittest added
Docs Changes: n/a
Release Notes: an item for bug fixes added.
Fixes https://github.com/envoyproxy/envoy/issues/22449

### CODE REVIEW

**Comment by phlax** [comment]
@soulxu as mentioned previously should this not be raised against `main` first ?

**Comment by soulxu** [comment]
@phlax yea, this is cherry pick from https://github.com/envoyproxy/envoy/pull/22622

**Comment by phlax** [comment]
cool - i just saw that this has landed on `main` already - apologies for noise (i have been off the last couple of days)

**Comment by soulxu** [comment]
> 

no worries! thanks for reminder this for me

File: changelogs/current.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@ minor_behavior_changes:
 
 bug_fixes:
 # *Changes expected to improve the state of the world and are unlikely to have negative effects*
+- area: listener
+  change: |
+    fixed a bug that doesn't handle of an update for a listener with IPv4-mapped address correctly, and that will lead to a memory leak.
+- area: transport_socket
+  change: |
+    fixed a bug that prevented the tcp stats to be retrieved when running on kernels different than the kernel where Envoy was built.
```
Comment (soulxu) [inline changelogs/current.yaml:16]: a merge mistake

**Comment by soulxu** [comment]
@phlax do you know why arm ci failed? it is missing some linux header file for iouring, that is strange.

**Comment by phlax** [comment]
/retest (for retest purposes)

i tried manually triggering and CI failed again for same reason

> @phlax do you know why arm ci failed?

no - seems odd - and unrelated to PR - also wondering why it didnt trigger this on `main`

**Review by mattklein123 — APPROVED** [review]

**Comment by phlax** [comment]
@soulxu searching on this bug and related to liburing and it seems its a kinda common problem with a few potential causes

- missing kernel headers
- incompatible kernel version
- problematic include paths

not sure why it is only failing on arm and only on this envoy version, and only once your patch is applied

im thinking we should probably debug to check kernel version and presence of headers etc

**Comment by soulxu** [comment]
/retest

Is https://github.com/envoyproxy/envoy/pull/22744 fix the CI issue? give a try

**Comment by phlax** [comment]
> Is https://github.com/envoyproxy/envoy/pull/22744 fix the CI issue? give a try

i think you will need to merge the branch update as i dont think ci merges as it does on `main`

**Comment by soulxu** [comment]
> > Is #22744 fix the CI issue? give a try
> 
> i think you will need to merge the branch update as i dont think ci merges as it does on `main`

just done, thanks!

**Review by phlax — APPROVED** [review]
lgtm, thanks @soulxu

**Pull request merged**